crates/store/src/genesis/config/mod.rs

```diff
@@ -0,0 +1,291 @@
+
+                        let now = chrono::Local::now();
+                        let now = now.format("%Y%m%d_%H%M%S");
+
+                        format!("{account_prefix}_account{idx:02}__{now}.mac")
+                    });
+                AccountFileWithName { name, account_file }
+            },
+        )
+    }
+}
+
+/// Specify a set of faucets and wallets with assets for easier test depoyments
+///
+/// Notice: Any faucet must be declared _before_ it's use in a wallet/regular account.
+#[derive(Debug, Clone, serde::Serialize, serde::Deserialize)]
+pub struct GenesisConfig {
+    version: u32,
+    timestamp: u32,
+    wallet: Vec<WalletConfig>,
+    faucet: Vec<FaucetConfig>,
+}
+
+impl GenesisConfig {
+    /// Read the genesis accounts from a toml formatted string
+    ///
+    /// Notice: It will generate the specified case during [`fn into_state`].
+    pub fn read_toml(toml_str: &str) -> Result<Self, Error> {
+        let me = toml::from_str::<Self>(toml_str)?;
+        Ok(me)
+    }
+
+    /// Convert the in memory representation into the new genesis state
+    ///
+    /// Notice: Generates keys and returns the secret keys, hence this is not sane to be used
+    /// for production environments. There, you want to generate the keys externally.
+    pub fn into_state(self) -> Result<(GenesisState, AccountSecrets), Error> {
+        let version = self.version;
+        let timestamp = self.timestamp;
+        let repr_accounts = self.wallet;
+        let repr_faucets = self.faucet;
+        let mut all_accounts = Vec::new();
+
+        // Every asset sitting in a wallet, has to reference a faucet for that asset
+        let mut faucets = HashMap::<String, AccountId>::new();
+
+        // Collect the generated secret keys for the test, so one can interact with those
+        // accounts/sign transactions
+        let mut secrets = Vec::<(Option<String>, Account, SecretKey, Word)>::new();
+
+        // First setup all the faucets
+        for FaucetConfig {
+            name,
+            symbol,
+            decimals,
+            max_supply,
+            storage_mode,
+            fungible,
+        } in repr_faucets
+        {
+            let mut rng = ChaCha20Rng::from_seed(rand::random());
+            let secret_key = SecretKey::with_rng(&mut get_rpo_random_coin(&mut rng));
+            let auth = RpoFalcon512::new(secret_key.public_key());
+            let init_seed: [u8; 32] = rng.random();
+
+            let token_symbol = TokenSymbol::new(&symbol)?;
+            let max_supply = Felt::try_from(max_supply).map_err(|_| {
+                Error::MaxSupplyExceedsFieldModulus { max_supply, modulus: Felt::MODULUS }
+            })?;
+
+            let account_type = if fungible {
+                AccountType::FungibleFaucet
+            } else {
+                AccountType::NonFungibleFaucet
+            };
+
+            if !fungible {
+                return Err(Error::UnsupportedValue {
```

Comment:
Is this implementable in the short term? Or is it fairly complex?

I guess I'm wondering if we aren't able to provide this fairly soon, should we bother adding the config for. And would these have a similar config interface, or should we physically split them into `FungibleFaucet` and `NonFungibleFaucet` with very different fields..

I'm not versed enough with the details here; @mmagician @drahnr thoughts?